[Bug] NPE when pushdown min/max to partitioned table which has a partition is all null

### Search before asking

- [x] I searched in the [issues](https://github.com/apache/paimon/issues) and found nothing similar.


### Paimon version

1.4-SNAPSHOT

### Compute Engine

Spark

### Minimal reproduce step

spark.sql("CREATE TABLE T (c1 INT, c2 LONG) PARTITIONED BY(day STRING)")

spark.sql("INSERT INTO T VALUES (1, 2, '2025-11-10')")
spark.sql("INSERT INTO T VALUES (null, 2, '2025-11-11')")

spark.sql("SELECT MIN(c1) FROM T").collect()

### What doesn't meet your expectations?

It doesn't work, an NPE is thrown.

[INTERNAL_ERROR] The Spark SQL phase optimization failed with an internal error. You hit a bug in Spark or the Spark plugins you use. Please, report this bug to the corresponding communities or vendors, and provide the full stack trace.
org.apache.spark.SparkException: [INTERNAL_ERROR] The Spark SQL phase optimization failed with an internal error. You hit a bug in Spark or the Spark plugins you use. Please, report this bug to the corresponding communities or vendors, and provide the full stack trace.
	at org.apache.spark.SparkException$.internalError(SparkException.scala:107)
	at org.apache.spark.sql.execution.QueryExecution$.toInternalError(QueryExecution.scala:536)
	at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:548)
	at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$1(QueryExecution.scala:219)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
	at org.apache.spark.sql.execution.QueryExecution.executePhase(QueryExecution.scala:218)
	at org.apache.spark.sql.execution.QueryExecution.optimizedPlan$lzycompute(QueryExecution.scala:148)
	...
Caused by: java.lang.NullPointerException
	at java.lang.Integer.compareTo(Integer.java:1216)
	at java.lang.Integer.compareTo(Integer.java:52)
	at org.apache.paimon.predicate.CompareUtils.compareLiteral(CompareUtils.java:31)
	at org.apache.paimon.spark.aggregate.MinEvaluator.update(AggFuncEvaluator.scala:63)
	at org.apache.paimon.spark.aggregate.LocalAggregator.$anonfun$update$2(LocalAggregator.scala:98)
	...

### Anything else?

_No response_

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

--- a/paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/aggregate/AggFuncEvaluator.scala
+++ b/paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/aggregate/AggFuncEvaluator.scala
@@ -59,7 +59,9 @@ case class MinEvaluator(idx: Int, dataField: DataField, evolutions: SimpleStatsE
 
   override def update(dataSplit: DataSplit): Unit = {
     val other = dataSplit.minValue(idx, dataField, evolutions)
-    // if (_result == null || (other != null && CompareUtils.compareLiteral(dataField.`type`(), _result, other) > 0)) {
+    if (other == null) {
+      return
+    }
     if (_result == null || CompareUtils.compareLiteral(dataField.`type`(), _result, other) > 0) {
       _result = other;
     }
@@ -81,7 +83,10 @@ case class MaxEvaluator(idx: Int, dataField: DataField, evolutions: SimpleStatsE
 
   override def update(dataSplit: DataSplit): Unit = {
     val other = dataSplit.maxValue(idx, dataField, evolutions)
-    if (_result == null || (other != null && CompareUtils.compareLiteral(dataField.`type`(), _result, other) < 0)) {
+    if (other == null) {
+      return
+    }
+    if (_result == null || CompareUtils.compareLiteral(dataField.`type`(), _result, other) < 0) {
       _result = other
     }
   }

--- a/paimon-spark/paimon-spark-ut/src/test/scala/org/apache/paimon/spark/sql/PushDownAggregatesTest.scala
+++ b/paimon-spark/paimon-spark-ut/src/test/scala/org/apache/paimon/spark/sql/PushDownAggregatesTest.scala
@@ -296,7 +296,7 @@ class PushDownAggregatesTest extends PaimonSparkTestBase with AdaptiveSparkPlanH
   }
 
   // https://github.com/apache/paimon/issues/6610
-  test("Push down aggregate - MIN/MAX of a column which in one partition is all null and the other is not") {
+  test("Push down aggregate: aggregate a column in one partition is all null and another is not") {
     withTable("T") {
       spark.sql("CREATE TABLE T (c1 INT, c2 LONG) PARTITIONED BY(day STRING)")
 

